autoscheduler/frontend/src/components/SchedulingPage/SchedulingPage.tsx

```diff
@@ -1,14 +1,24 @@
 import * as React from 'react';
-import { useDispatch } from 'react-redux';
-import { RouteComponentProps } from '@reach/router';
+import { useDispatch, useSelector } from 'react-redux';
+import { RouteComponentProps, navigate } from '@reach/router';
 import Schedule from './Schedule/Schedule';
 import * as styles from './SchedulingPage.css';
 import ConfigureCard from './ConfigureCard/ConfigureCard';
 import SchedulePreview from './SchedulePreview/SchedulePreview';
 import CourseSelectColumn from './CourseSelectColumn/CourseSelectColumn';
 import setTerm from '../../redux/actions/term';
+import { RootState } from '../../redux/reducer';
 
 const SchedulingPage: React.FC<RouteComponentProps> = (): JSX.Element => {
+  // get term from redux state
+  const termForRedir = useSelector<RootState, string>((state) => state.term);
```

Comment:
So honestly I think we simplified this a bit too much when we explained this. While this does technically fix the issue we had, it also ignores the fact that we have `sessions/get_last_term`, which restores what term you had picked last from your saved session. Moreover, because the Redux gets completely wiped on refreshing, term will _always_ be null on refresh.

What this should actually do is wait for the `sessions/get_last_term` response (or more over go in the response handling of it), and if there is no saved term, then should it redirect you to the landing page. You can instead remove this `useSelector` and such, and just add the check + `navigate` around line 29 in this file.

Reply:
![image](https://user-images.githubusercontent.com/49422137/95690472-6081f000-0bd5-11eb-962a-8ed2a88e575f.png)
Is this more what you wanted?
This does reintroduce the flashing problem but idt we can stop that since its on a short delay now?
- If you want I can try to move that code to a pre-render hook (getDerivedStateFromProps()?)

Reply:
![image](https://user-images.githubusercontent.com/49422137/95794100-5f6bc400-0ca4-11eb-8082-4d9598a315fd.png)
![image](https://user-images.githubusercontent.com/49422137/95794067-4bc05d80-0ca4-11eb-8c81-c1165c3c06c6.png)
Do these look good?